pywbemtools/pywbemcli/_cmd_class.py

```diff
@@ -823,58 +823,79 @@ def cmd_class_find(context, classname_glob, options):
         raise_pywbem_error_exception(er)
 
 
-def cmd_class_tree(context, classname, options):
+def get_class_hiearchy(conn, classname, namespace, superclasses=None):
     """
-    Execute the command to enumerate classes from the top or starting at the
-    classname argument. Then format the results to be displayed as a
-    left-justified tree using the asciitree library.
-    The --superclasses option determines if the superclass tree or the
-    subclass tree is displayed.
+    Get the class hiearchy from the server, either the superclasses
+    associated with a class or the subclasses.  If getting subclasses
+    the classname parameter may be None which requests that the complete
+    class hiearchy be retrieved..
+
+      Parameters:
+        conn(TODO):
+
+        classname(:term:`string):
+            classname if the tree is to be initiated from
+            within the class hiearchy. May be None
+
+        namespace(:term:`string):
+          Namespace to use to acquire classes from WBEM server
+
+      Returns:
+
+      Raises:
+        CIM_Error:
     """
 
     # TODO FUTURE: Sort out how we handle output format with tree output.
     try:
-        if options['superclasses']:
+        if superclasses:
             if classname is None:
                 raise click.ClickException('CLASSNAME argument required for '
                                            '--superclasses option')
```

Comment:
It seems to me that such a handling of missing command arguments should be done by the caller, and not in this retrieval function.

Reply:
Do not understand this comment.  The code is

if superclasses:

It is not there as an assertion but as a test whether the user wanted superclasses or subclasses.

**DISCUSSION**

Reply:
That was also a good change, but I meant the whole check on lines 858 to 861 (in the changed version):
```
        if superclasses:
            if classname is None:
                raise click.ClickException('CLASSNAME argument required for '
                                           '--superclasses option')
```
That check is already done in cmd_class_tree() which is the only caller.
So the check here could be changed to become an assertion:
```
        if superclasses:
            assert classname is not None
```

Reply:
Whoops, missed that duplication.  Fixed. DONE